netbox_netprod_importer/devices_list.py

```diff
@@ -28,3 +31,57 @@ def parse_devices_yaml_def(devices_yaml, creds=None):
                     "Cannot connect to device %s: %s", hostname, e
                 )
     return devices
+
+
+def parse_filter_yaml_def(filter_yaml, creds=None):
+    netbox_api = NetboxAPI(**get_config()["netbox"])
+    devices = {}
+    with open(filter_yaml) as filter_yaml_str:
+        q_list = []
+        yml = yaml.safe_load(filter_yaml_str)
+        for fil, props in tqdm(yml["filter"].items()):
+            if props is not None:
+                if type(props) is list:
+                    q_list.append(fil + "=" +
+                                  ("&" + fil + "=").join(str(x) for x in props)
+                                  )
+                else:
+                    q_list.append(fil + "=" + str(props))
+        logger.info("Resultant filter %s", "&".join(q_list))
+        platforms_js = netbox_api.get("dcim/platforms")
+        platforms = {}
```

Comment:
As it is only existing in this scope of the function, I would transform `platform` to a dict of tuple, containing `(napalm_driver, napalm_args)`, in order to avoid having the `platform_args` variable.

So then you can do: `platforms[platform["id"]] = (platform["napalm_driver"], platform["napalm_args"])`

Reply:
I do not like working with tuples (not informatively), made through dictionaries.

Reply:
Then you can do a dict of dict `{"driver": …, "args":…}` ?